resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -73,6 +76,30 @@ private[spark] class Client(
   private val driverJavaOptions = sparkConf.get(
     org.apache.spark.internal.config.DRIVER_JAVA_OPTIONS)
 
+  // create resource of kind - SparkJob representing the deployed spark app
+  private val statusMap = Map(
```

Comment:
Can we use a more specific type than a map? If we always want the same fields, use a case class with the specific fields enumerated.